### ISSUE TITLE
daemon.js keeps throwing errors

### ISSUE THREAD

**Issue by zocke1r** [alainbryden/bitburner-scripts#88]
Most of the time when a server gets deleted i get this error 3 times  'daemon.js: ERROR: Failed to find server by name |DELIMITER|home|DELIMITER|daemon.js|DELIMITER|getServerSecurityLevel: Invalid hostname: .'

**Comment by alainbryden** [comment]
Can I get a screenshot?

**Comment by zocke1r** [comment]
![image](https://user-images.githubusercontent.com/12860183/163673117-0b8f1eca-fb32-4421-9417-1910da28c085.png)

**Comment by alainbryden** [comment]
I wonder if this is a regression in the game? Usually when you delete a server, subsequent attempts to use the server result in the error "Invalid hostname: oldServerName" - and I can automatically detect this and handle the deleted server.

But the error message appear to contain no hostname (unless you manually bought a server and named it " " (empty string)?

Perhaps you can send me your save file, and I can turn off error handling and get to the bottom of this.

**Comment by alainbryden** [comment]
Actually, it kind of looks like somehow a null/undefined/empty string is getting into your list of hosts somewhere, I'm not sure it has to do with deleting a server. So a game save to see what's happening would be excellent.

**Comment by Evanito** [comment]
One of the hosts is named `.` btw, I can +1 encountering this error though not nearly as often

**Comment by alainbryden** [comment]
Hmm, you may have encountered a bug in the game itself, "." Is the name of
an actual server in the game, you may have confused it thoroughly. My
suggestion is try to delete the host, stop all scripts, and restart you
game 🤷

**Comment by alainbryden** [comment]
Alright, I think I got to the bottom of this.

The error message was misleading, it was caused by passing a -1 to the "length" of the substring function, which insanely causes it to return the whole string up to "start". The -1 was because the delimiter "<br>" has changed (or at least, isn't used in all error messages about an "Invalid hostname:")

I know handle a few scenarios and should be able to better detect deleted servers.
